### PR TITLE
Start a simple github actions CI

### PR DESCRIPTION
**Pull request by hzeller**
Running on major platforms.

I'd like to contribute a starting point for a CI that can run on pushes and pull requests, which is in particular useful to make sure that things continue working on all platforms.

Note, the Windows run actually fails right now and it looks like it might actually be worthwhile looking into that. One test looks like a CR to CRLF mismatch. And there is some death test message that doesn't fit. I'll leave that to the googletest team to work out.

### CODE REVIEW

**Comment by hzeller** [comment]
It might be needed to enable actions for this repository before they can run.

**Comment by dinord** [comment]
We've decided to accept this PR, and start using GitHub actions as a best-effort community contribution CI that can be triggered by open source users, and not just Googlers.
For now, we'll still keep using Kokoro CI as the source of truth for breakages.

File: .github/workflows/gtest-ci.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+name: ci
+
+on:
+  push:
+  pull_request:
```
Comment (danielkrupinski) [inline .github/workflows/gtest-ci.yml:5]: `on: [push, pull_request]` is also an option

File: .github/workflows/gtest-ci.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  Windows:
+    runs-on: windows-latest
+    steps:
+
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+
+    - name: Tests
+      run: bazel test --test_output=errors //...
```
Comment (danielkrupinski) [inline .github/workflows/gtest-ci.yml:40]: Duplications can be avoided by using a [strategy matrix](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix):
```
jobs:
  build:
    name: ${{ matrix.name }}
    runs-on: ${{ matrix.os }}
    strategy:
      matrix:
        os: [ubuntu-latest, macos-latest, windows-latest]
        include:
          - os: ubuntu-latest
            name: Linux
          - os: macos-latest
            name: MacOs
          - os: windows-latest
            name: Windows

    steps:
    - uses: actions/checkout@v2
      with:
        fetch-depth: 0
    - name: Tests
      run: bazel test --test_output=errors //...

**Pull request merged**